darts/explainability/shap_explainer.py

```diff
@@ -698,13 +700,33 @@ def _create_regression_model_shap_X(
             past_covariates=past_covariates,
             future_covariates=future_covariates,
             lags=lags_list,
+            concatenate=False,
             lags_past_covariates=lags_past_covariates_list if past_covariates else None,
             lags_future_covariates=lags_future_covariates_list
             if future_covariates
             else None,
         )
         # Remove sample axis:
-        X = X[:, :, 0]
+        for i, X_i in enumerate(X):
+            X[i] = X_i[:, :, 0]
+
+
+        X, _ = add_static_covariates_to_lagged_data(X, target_series, uses_static_covariates=self.model.uses_static_covariates)
+
+        # generate and store the lagged components names (for feature importance analysis)
+        self.model._lagged_feature_names, _ = create_lagged_component_names(
+            target_series=target_series,
+            past_covariates=past_covariates,
+            future_covariates=future_covariates,
+            lags=self.model.lags.get("target"),
+            lags_past_covariates=self.model.lags.get("past"),
+            lags_future_covariates=self.model.lags.get("future"),
+            output_chunk_length=self.model.output_chunk_length,
+            concatenate=False,
+            use_static_covariates=self.model.uses_static_covariates,
+        )
+
+        X = np.concatenate(X, axis=0)
```

Comment:
This should not be necessary if the operation are performed on the array directly (instead of the for loop)